packages/salesforce-adapter/src/filters/flows_filter.ts

```diff
@@ -47,62 +47,103 @@ import { SalesforceRecord } from '../client/types'
 
-const getActiveFlowVersionIdByApiName = async ({
+const getFlowVersionsByApiName = async ({
   client,
   flowDefinitions,
   chunkSize,
 }: {
   client: SalesforceClient
   flowDefinitions: InstanceElement[]
   chunkSize: number
-}): Promise<Record<string, string>> => {
-  const flowDefinitionsIds = flowDefinitions.map(flow => flow.value[INTERNAL_ID_FIELD])
+}): Promise<Map<string, FlowVersionProperties[]>> => {
+  const flowDefinitionsIdsToApiNames = Object.fromEntries(
+    flowDefinitions.map(flow => [flow.value[INTERNAL_ID_FIELD], flow.value[INSTANCE_FULL_NAME_FIELD]]),
+  )
   const records = _.flatten(
     await Promise.all(
-      _.chunk(flowDefinitionsIds, chunkSize).map(async chunk => {
-        const query = `SELECT Id, ApiName, ActiveVersionId FROM FlowDefinitionView WHERE Id IN ('${chunk.join("','")}')`
-        return (await toArrayAsync(await client.queryAll(query))).flat()
+      _.chunk(Object.keys(flowDefinitionsIdsToApiNames), chunkSize).map(async chunk => {
+        const query = `SELECT Id, DefinitionId, VersionNumber, Status, CreatedDate, CreatedBy.Name, LastModifiedDate, LastModifiedBy.Name FROM Flow WHERE DefinitionId IN ('${chunk.join("','")}')`
+        return (await toArrayAsync(await client.queryAll(query, true))).flat()
       }),
     ),
   )
-  const [validRecords, invalidRecords] = _.partition(records, isFlowDefinitionViewRecord)
+  const [validRecords, invalidRecords] = _.partition(records, isFlowVersionRecord)
   if (invalidRecords.length > 0) {
     log.error(
-      'Some FlowDefinitionView records are invalid. Records are: %s',
+      'Some Flow version records are invalid. Records are: %s',
       inspectValue(invalidRecords, { maxArrayLength: 10 }),
     )
     if (invalidRecords.length > 10) {
-      log.trace(
-        'All Invalid FlowDefinitionView records are: %s',
-        inspectValue(invalidRecords, { maxArrayLength: null }),
-      )
+      log.trace('All invalid Flow version records are: %s', inspectValue(invalidRecords, { maxArrayLength: null }))
     }
   }
-  return validRecords.reduce<Record<string, string>>((acc, record) => {
-    if (record.ActiveVersionId !== null) {
-      acc[record.ApiName] = record.ActiveVersionId
-    }
-    return acc
-  }, {})
+  return validRecords.reduce<Map<string, FlowVersionProperties[]>>(
+    (acc, record) => {
+      const apiName = flowDefinitionsIdsToApiNames[record.DefinitionId]
+      if (apiName === undefined) {
+        log.warn('Got flow version with unrecognized flow definition ID: %s', inspectValue(record))
+        return acc
+      }
+      acc.get(apiName)?.push({
+        id: record.Id,
+        version: record.VersionNumber,
+        status: record.Status,
+        createdByName: record.CreatedBy.Name,
+        createdDate: record.CreatedDate,
+        lastModifiedByName: record.LastModifiedBy.Name,
+        lastModifiedDate: record.LastModifiedDate,
+      })
+      return acc
+    },
+    new DefaultMap(() => []),
+  )
```

Comment:
Why the downcast to `Map`? Please note `get` of `DefaultMap` will never return undefined.
```suggestion
  return validRecords.reduce(
    (acc, record) => {
      const apiName = flowDefinitionsIdsToApiNames[record.DefinitionId]
      if (apiName === undefined) {
        log.warn('Got flow version with unrecognized flow definition ID: %s', inspectValue(record))
        return acc
      }
      acc.get(apiName).push({
        id: record.Id,
        version: record.VersionNumber,
        status: record.Status,
        createdByName: record.CreatedBy.Name,
        createdDate: record.CreatedDate,
        lastModifiedByName: record.LastModifiedBy.Name,
        lastModifiedDate: record.LastModifiedDate,
      })
      return acc
    },
    new DefaultMap<string, FlowVersionProperties[]>(() => []),
  )
```